master/internal/command/command.go

```diff
@@ -40,10 +38,39 @@ type readinessCheck func(sproto.ContainerLog) bool
 // should stop and garbage collect its state.
 type terminateForGC struct{}
 
-// commandOwner describes the owner of a command.
-type commandOwner struct {
-	ID       model.UserID `json:"id"`
-	Username string       `json:"username"`
+func generateServiceAddress(taskID string) string {
+	return fmt.Sprintf("/proxy/%s/", taskID)
+}
+
+func createGenericCommandActor(
+	ctx *actor.Context,
+	db *db.PgDB,
+	spec tasks.GenericCommandSpec,
+	readinessCheck map[string]readinessCheck,
+) error {
+	taskID := sproto.NewTaskID()
+
+	serviceAddress := generateServiceAddress(string(taskID))
+
+	cmd := &command{
+		db:              db,
+		readinessChecks: readinessCheck,
+
+		GenericCommandSpec: spec,
+
+		taskID:         taskID,
+		serviceAddress: &serviceAddress,
```

Comment:
to me it makes more sense to me to have the command manage creating the task/task_id/existence of a service address of the existence of some proxy spec or something. we dont even use this to define the service address, we look for the exposed genericspec.port and open a proxy and assume the proxy is started as we derive here?

i'd prefer: proxyspec.enabled bool + proxyspec.port results in us sending a proxy.Registry and that ask should return the service address, and the service address is nil until created.

Reply:
Let's figure out proxySpec in a followup PR.